transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoopGroup.java

```diff
@@ -46,105 +46,57 @@ public IOUringEventLoopGroup(int nThreads) {
+                                  int ringSize, RejectedExecutionHandler rejectedExecutionHandler) {
+        this(nThreads, executor, chooserFactory, ringSize, rejectedExecutionHandler, null);
     }
 
-    public IOUringEventLoopGroup(int nThreads, Executor executor, EventExecutorChooserFactory chooserFactory,
-                                 SelectStrategyFactory selectStrategyFactory) {
-        super(nThreads, executor, chooserFactory, 0, selectStrategyFactory, RejectedExecutionHandlers.reject());
-    }
-
-    public IOUringEventLoopGroup(int nThreads, Executor executor, EventExecutorChooserFactory chooserFactory,
-                                 SelectStrategyFactory selectStrategyFactory,
-                                 RejectedExecutionHandler rejectedExecutionHandler) {
-        super(nThreads, executor, chooserFactory, 0, selectStrategyFactory, rejectedExecutionHandler);
-    }
-
-    public IOUringEventLoopGroup(int nThreads, Executor executor, EventExecutorChooserFactory chooserFactory,
-                                 SelectStrategyFactory selectStrategyFactory,
-                                 RejectedExecutionHandler rejectedExecutionHandler,
-                                 EventLoopTaskQueueFactory queueFactory) {
-        super(nThreads, executor, chooserFactory, 0,
-              selectStrategyFactory, rejectedExecutionHandler, queueFactory);
-    }
-
-    /**
-     * @deprecated This method will be removed in future releases, and is not guaranteed to have any impacts.
-     */
-    @Deprecated
-    public void setIoRatio(int ioRatio) {
-        if (ioRatio <= 0 || ioRatio > 100) {
-            throw new IllegalArgumentException("ioRatio: " + ioRatio + " (expected: 0 < ioRatio <= 100)");
-        }
+    private IOUringEventLoopGroup(int nThreads, Executor executor, EventExecutorChooserFactory chooserFactory,
+                                  int ringSize, RejectedExecutionHandler rejectedExecutionHandler,
+                                  EventLoopTaskQueueFactory queueFactory) {
+        super(nThreads, executor, chooserFactory, ringSize, rejectedExecutionHandler, queueFactory);
     }
 
     //Todo
     @Override
-    protected EventLoop newChild(Executor executor, Object... args) throws Exception {
-     //EventLoopTaskQueueFactory queueFactory = args.length == 4? (EventLoopTaskQueueFactory) args[3] : null;
-//        return new IOUringEventLoop(this, executor, (Integer) args[0],
-//                ((SelectStrategyFactory) args[1]).newSelectStrategy(),
-//                (RejectedExecutionHandler) args[2], queueFactory);
-
-        return new IOUringEventLoop(this, executor, false);
+    protected EventLoop newChild(Executor executor, Object... args) {
+        if (args.length != 3) {
+            throw new IllegalArgumentException("Illegal amount of extra arguments");
+        }
+        int ringSize = (Integer) args[0];
+        if (ringSize == 0) {
```

Comment:
nit: what about negative values?

Reply:
it will fail when we try to create the ring I guess... But let me add some validation code.